engine/consensus/matching/engine.go

```diff
@@ -535,30 +530,21 @@ RES_LOOP:
 			return nil, fmt.Errorf("could not retrieve block: %w", err)
 		}
 
-		// Retrieve parent result / skip if parent result still unknown:
-		// Before we store a result into the incorporatedResults mempool, we store it in resultsDB.
-		// I.e. resultsDB contains a superset of all results stored in the mempool. Hence, we only need
-		// to check resultsDB. Any result not in resultsDB cannot be in incorporatedResults mempool.
-		previousID := incorporatedResult.Result.PreviousResultID
-		previous, err := e.resultsDB.ByID(previousID)
-		if errors.Is(err, storage.ErrNotFound) {
-			e.log.Debug().Msg("skipping sealable result with unknown previous result")
-			continue
-		}
-		if err != nil {
-			return nil, fmt.Errorf("could not get previous result: %w", err)
-		}
-
-		// check sub-graph
-		if block.ParentID != previous.BlockID {
+		// TODO: move block (header) to the IncorporatedResult struct,
+		// get rid of the above non-existing error case
+		err = validateResult(e.resultsDB, block, incorporatedResult)
+		if engine.IsInvalidInputError(err) {
 			_ = e.incorporatedResults.Rem(incorporatedResult)
 			e.log.Warn().
-				Str("block_parent_id", block.ParentID.String()).
-				Str("previous_result_block_id", previous.BlockID.String()).
+				Err(err).
 				Msg("removing result with invalid sub-graph")
 			continue
 		}
 
+		if err != nil {
+			return nil, fmt.Errorf("could not validate result: %w", err)
+		}
+
```

Comment:
As Martin pointed out:
> This is already done by the module.ReceiptValidator.

Removing this code would address some of the technical debt listed in issue https://github.com/dapperlabs/flow-go/issues/5137; specifically point 3: 
> * [ ] `matching.Engine` duplicates logic that is now [`ReceiptValidator`](https://github.com/onflow/flow-go/blob/3369dbd1f1e965fae2556f84d677b17297120a05/module/validation/receipt_validator.go)